server/workers/shareScrollWorker.ts

```diff
@@ -0,0 +1,548 @@
+      const flairImage = sharp(flairPath)
+        .greyscale()
+        .threshold(255)
+        .composite([
+          {
+            input: Buffer.from([255, 255, 255, Math.ceil(255 / 5)]),
+            raw: {
+              width: 1,
+              height: 1,
+              channels: 4,
+            },
+            tile: true,
+            blend: 'dest-in',
+          },
+          { input: flairPath, left: -1, top: -1 },
+        ])
+        .png();
+      const { height: flairHeight } = await flairImage.metadata();
+      composites.push({
+        input: await flairImage.toBuffer(),
+        left: 118 + usernameWidth + 4,
+        top: 16 - Math.floor((flairHeight ?? 0) / 2),
+      });
+    }
+
+    const statText = (statName: string, statNumber: number) => `
+      <tspan fill="#dff6f5">${statName}:</tspan> 
+      <tspan fill="#f2bd59">${statNumber.toLocaleString()}</tspan>
+    `;
+
+    // stats
+    composites.push({
+      input: await textToPng(
+        statText('Dragons', dragonCount),
+        '8px Nokia Cellphone FC',
+        ''
+      ),
+      left: 118,
+      top: 28,
+    });
+    composites.push({
+      input: await textToPng(
+        statText('Growing', growingCount),
+        '8px Nokia Cellphone FC',
+        ''
+      ),
+      left: 118,
+      top: 40,
+    });
+    composites.push({
+      input: await textToPng(
+        statText('Weekly Clicks', weeklyClicks),
+        '8px Nokia Cellphone FC',
+        ''
+      ),
+      left: 200,
+      top: 28,
+    });
+    composites.push({
+      input: await textToPng(
+        statText('All-time Clicks', allTimeClicks),
+        '8px Nokia Cellphone FC',
+        ''
+      ),
+      left: 200,
+      top: 40,
+    });
+
+    console.log(
+      `Banner stats generated in ${(performance.now() - startTime).toFixed(2)}ms`
+    );
+    const bannerBuffer = await compositeImage
+      .composite(composites)
+      .png()
+      .toBuffer();
+    return {
+      bannerBuffer,
+      bannerHeight: BANNER_HEIGHT,
+      bannerWidth: BANNER_WIDTH,
+    };
+  } catch (error) {
+    console.error('Error in getBannerBase:', error);
+    throw error;
+  }
+}
+
+//
+
+async function getDragonStrip(dragonIds: string[]) {
+  // it's fine for there to be no growing things,
+  // it just means the carousel will be empty.
+  if (dragonIds.length === 0) {
+    return {
+      stripBuffer: null,
+      stripWidth: 0,
+      stripHeight: 0,
+    };
+  }
+
+  try {
+    const startTime = performance.now();
+    console.log('Generating growing images...');
+    const dragonBuffers = await Promise.all(
+      dragonIds.map(async (dragonId) => {
+        const response = await fetch(
+          `https://dragcave.net/image/${dragonId}.gif`
+        );
+        if (!response.ok) {
+          throw new Error(
+            `Failed to fetch dragon image ${dragonId}: ${response.statusText}`
+          );
+        }
+        const arrayBuffer = await response.arrayBuffer();
```

Comment:
I want to _say_ fetch already throws its own error I think. unless you need to see this specifically...?

Reply:
The fetch can still return a not-okay response without erroring, like a 404. Unless I'm missing something obvious?

Reply:
reasonable